test/library/standard/List/perf/listBenchmark1.chpl

```diff
@@ -0,0 +1,172 @@
+//
+// A performance test benchmarking common list operations.
+//
+
+private use HaltWrappers;
+private use List;
+private use Random;
+private use Time;
+
+type byte = int(8);
+type testList = list(byte, false);
+
+config param isPerformanceTest: bool = false;
+config const n0: int = 50000;
+
+const n1: int = n0 / 1000;
+const seed = 314159;
+const rand = createRandomStream(eltType=int, seed=seed, parSafe=false);
+
+proc createList(size: int) {
+  var result: testList;
+  for i in 1..size do result.append(0:int(8));
+  return result;
+}
+
+proc generateNoise() {
+  const size = if isPerformanceTest then n1 else n0;
+  var lst1 = createList(size);
+  var lst2 = createList(size);
+
+  for (a, b) in zip(lst1, lst2) do a += b % 127;
+
+  lst1.clear();
+  lst2.clear();  
+}
+
+class Test {
+  proc name(): string { pureVirtualMethodHalt(); }
+  proc setup() {}
+  proc test() { pureVirtualMethodHalt(); }
+
+  proc run(): real {
+    var tmr = new Timer();
+
+    setup();
+
+    tmr.start();
+    test();
+    tmr.stop();
+
+    return tmr.elapsed();
+  }
+
+  proc output() {
+    // Run tests even if we aren't `-performance`.
+    const time = this.run();
+
+    const msg = if isPerformanceTest
+      then this.name() + " " + time:string
+      else this.name();
+    writeln(msg);
+  }
+}
+  
+class AppendFromEmpty: Test {
+  var _lst: testList;
+
+  override proc name() return "AppendFromEmpty";
+  override proc setup() { _lst.clear(); }
+
+  override proc test() {
+    assert(_lst.isEmpty());
+    for i in 1..n0 do _lst.append((i % 127):byte);
+  }
+}
+
+class InsertFront: Test {
+  // Use a smaller value for N because InsertFront is a O(n**2) operation.
+  var _lst: testList;
+ 
+  override proc name() return "InsertFront";
+  override proc setup() { _lst = createList(1); }
+
+  override proc test() {
+    assert(_lst.size == 1);
+    while _lst.size < n1 do _lst.insert(1, (_lst.size & 127):byte);
```

Comment:
I might be tempted to do:

```chpl
for 1..n1-1 do 
  _lst.insert(1, (_lst.size & 127):byte); 
assert(_lst.size == n1);
```

Though it probably doesn't matter one way or the other.